eZ/Publish/API/Repository/Tests/SearchServiceLocationTest.php

```diff
@@ -1037,6 +1037,100 @@ public function testMapLocationDistanceWithCustomFieldSort()
         );
     }
 
+    /**
+     * Test for the findLocations() method.
+     *
+     * @see \eZ\Publish\API\Repository\SearchService::findLocations()
+     */
+    public function testVisibilityCriterionWithHiddenContent()
+    {
+        $repository = $this->getRepository();
+        $contentTypeService = $repository->getContentTypeService();
+        $contentType = $contentTypeService->loadContentTypeByIdentifier('folder');
+
+        $contentService = $repository->getContentService();
+        $locationService = $repository->getLocationService();
+        $searchService = $repository->getSearchService();
+
+        $testRootContentCreate = $contentService->newContentCreateStruct($contentType, 'eng-US');
+        $testRootContentCreate->setField('name', 'Root for test');
+
+        $rootContent = $contentService->createContent(
+            $testRootContentCreate,
+            [
+                $locationService->newLocationCreateStruct(
+                    $this->generateId('location', 2)
+                ),
+            ]
+        );
+
+        $publishedRootContent = $contentService->publishVersion($rootContent->versionInfo);
+
+        $contentCreate = $contentService->newContentCreateStruct($contentType, 'eng-US');
+        $contentCreate->setField('name', 'To Hide');
+
+        $content = $contentService->createContent(
+            $contentCreate,
+            [
+                $locationService->newLocationCreateStruct(
+                    $publishedRootContent->contentInfo->mainLocationId
+                ),
+            ]
+        );
+        $publishedContent = $contentService->publishVersion($content->versionInfo);
+
+        $childContentCreate = $contentService->newContentCreateStruct($contentType, 'eng-US');
+        $childContentCreate->setField('name', 'Invisible Child');
+
+        $childContent = $contentService->createContent(
+            $childContentCreate,
+            [
+                $locationService->newLocationCreateStruct(
+                    $publishedContent->contentInfo->mainLocationId
+                ),
+            ]
+        );
+        $contentService->publishVersion($childContent->versionInfo);
+        $this->refreshSearch($repository);
+
+        $query = new LocationQuery([
+            'query' => new Criterion\LogicalAnd([
+                new Criterion\Visibility(
+                    Criterion\Visibility::VISIBLE
+                ),
+                new Criterion\Subtree(
+                    '/1/2/' . $publishedRootContent->contentInfo->mainLocationId . '/'
```

Comment:
You probably should load this Location and and use `Location::pathString` property